frontend/src/components/GenerateSchedulesUnitAndTimePicker.js

```diff
@@ -0,0 +1,184 @@
+import React,{useState} from 'react';
+import {Card, Form, Row, Button, Col, FormLabel, FormGroup, Modal} from 'react-bootstrap';
+import {Select} from "@material-ui/core";
+import InputLabel from '@material-ui/core/InputLabel';
+import { makeStyles } from '@material-ui/core/styles';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import FormControl from '@material-ui/core/FormControl';
+import Container from '@material-ui/core/Container';
+
+import {Label} from "@material-ui/icons";
+const useStyles = makeStyles((theme) => ({
+    formControl: {
+        margin: theme.spacing(1),
+        minWidth: 120,
+    },
+    selectEmpty: {
+        marginTop: theme.spacing(2),
+    },
+    container: {
+        display: 'flex',
+        flexWrap: 'wrap',
+    },
+    textField: {
+        marginLeft: theme.spacing(1),
+        marginRight: theme.spacing(1),
+        width: 200,
+    },
+}));
+
+
+export default function GenerateSchedulesUnitAndTimePicker() {
```

Comment:
(note for later reference) setting minimum units, maximum units, and time constraints will be passed as props